File: 190820/7964.c
# include <stdio.h>

int main(void) {
	int T;
	scanf_s("%d", &T);
	int N = 0;
	int M = 0;
	int arr[10000];
	int position;
	int remain;
	int input;
	int count;
	int l;
	int m;
	for (int test_case = 1; test_case <= T; test_case++) {
		position = -1;
		remain = 0;
		scanf_s("%d", &N);
		scanf_s("%d", &M);
		for (int i = 0; i < N; i++) {
			scanf_s("%d", &input);
			if (input) {
				arr[remain] = i;
				remain++;
			}
		}
		arr[remain] = N;
		count = 0;
		l = 0;
		m = 1;
		while (position < N) {
			if (remain) {
				if ((position + 1 <= arr[l]) && (arr[l] <= position + M)) {
					position = arr[l];
					if (l == remain) {
						remain = 0;
						m = 1;
					}
					else {
						l++;
						m = 0;
					}
				}
				else {
					m = 1;
				}
			}
			if (m) {
				position += M;
				if (position < N)	count++;
			}
			if (position + M > N) {
				break;
			}
		}
		printf("#%d %d\n", test_case, count);
	}
	return 0;
}

File: 190820/7964.py
import sys
sys.stdin = open('7964.txt', 'r')

for test_case in range(1, int(input()) + 1):
    N, M = tuple(map(int, input().split()))
    arr = input().split()
    position = -1
    count = 0
    while position < N:
        for i in range(position + M, position, -1):
            if arr[i] == '1':
                position = i
                break
        else:
            position += M
            count += 1
        if position + M >= N:
            break
    print(count)
